tests/Unit/Domain/Creators/Test_ShapeMapOptions.cpp

```diff
@@ -215,46 +215,78 @@ void test_funcs(const gsl::not_null<Generator*> generator) {
-      // Make sure they were restricted properly
-      CHECK(gsl::at(shape_funcs, i).size() == expected_size);
+      ylm::SpherepackIterator iter{l_max, l_max};
+      ylm::SpherepackIterator file_iter{file_l_max, file_l_max};
+      for (size_t i = 0; i < shape_funcs.size(); i++) {
+        const size_t expected_size =
+            ylm::Spherepack::spectral_size(l_max, l_max);
+        // Make sure they were restricted properly
+        CHECK(gsl::at(shape_funcs, i).size() == expected_size);
 
-      // Loop pointwise so we only check the coefficients that matter. Also
-      for (size_t l = 0; l <= l_max; l++) {
-        for (int m = -static_cast<int>(l); m <= static_cast<int>(l); m++) {
-          CAPTURE(l);
-          CAPTURE(m);
-          const double expected_value =
-              l < 2 ? 0.0
-                    : -1.0 * gsl::at(strahlkorpers, i)
-                                 .coefficients()[file_iter.set(l, m)()];
-          CHECK(gsl::at(shape_funcs, i)[iter.set(l, m)()] == expected_value);
+        // Loop pointwise so we only check the coefficients that matter
+        for (size_t l = 0; l <= l_max; l++) {
+          for (int m = -static_cast<int>(l); m <= static_cast<int>(l); m++) {
+            CAPTURE(l);
+            CAPTURE(m);
+            const double expected_value =
+                l < 2 ? 0.0
+                      : -1.0 * gsl::at(strahlkorpers, i)
+                                   .coefficients()[file_iter.set(l, m)()];
+            CHECK(gsl::at(shape_funcs, i)[iter.set(l, m)()] == expected_value);
+          }
         }
       }
+      CHECK(size_funcs == std::array{DataVector{1.1}, DataVector{2.2},
+                                     DataVector{3.3}, DataVector{0.0}});
+    }
+
+    // Already checked that shape funcs are correct. Here just check that size
+    // funcs were automatically set to correct values
+    {
+      const auto shape_map_options = TestHelpers::test_creation<
+          domain::creators::time_dependent_options::ShapeMapOptions<
+              false, domain::ObjectLabel::None>>(
+          "LMax: 8\n"
+          "InitialValues:\n"
+          "  H5Filename: TotalEclipseOfTheHeart.h5\n"
+          "  SubfileNames:\n"
+          "    - Ylm_coefs\n"
+          "    - dt_Ylm_coefs\n"
+          "  MatchTime: 1.7\n"
+          "  MatchTimeEpsilon: Auto\n"
+          "  SetL1CoefsToZero: True\n"
+          "SizeInitialValues: Auto");
+
+      const auto [shape_funcs, size_funcs] =
+          domain::creators::time_dependent_options::
+              initial_shape_and_size_funcs(shape_map_options, inner_radius);
+
+      CHECK(size_funcs ==
+            std::array{DataVector{-1.0 * strahlkorpers[0].coefficients()[0] *
+                                  sqrt(0.5 * M_PI)},
+                       DataVector{-1.0 * strahlkorpers[1].coefficients()[0] *
+                                  sqrt(0.5 * M_PI)},
+                       DataVector{0.0}, DataVector{0.0}});
```

Comment:
Also check zero coeffs of shape funcs.

Maybe a stronger test would be to write out a strahlkorper, load it back in, and check that the xi=-1 logical coord of blocks abutting the excision map to the cartesian coords of the strahlkorper. I added such a test for the sphere and BBH domains for the Kerr-horizon-conforming case, so you can copy most of that. That test would catch any bugs with factors of pi or minus signs or anything like that.